[Profiler] Pull helper methods into dedicated file. (And start `torch/csrc/profiler` folder.

robieta opened the pull request:
Summary: One thing that I've found as I optimize profier is that there's a lot of intermingled code, where the kineto profiler relies on the legacy (autograd) profiler for generic operations. This made optimization hard because I had to manage too many complex dependencies. (Exaserbated by the USE_KINETO #ifdef's sprinkled around.) This PR is the first of several to restructure the profiler(s) so the later optimizations go in easier.

Test Plan: Unit tests

Reviewed By: aaronenyeshi

Differential Revision: D32671972

albanD commented on torch/csrc/profiler/util.h:
```diff
@@ -0,0 +1,141 @@
+    const std::unordered_map<std::string, c10::IValue>& extra_args);
+
+} // namespace profiler
+} // namespace autograd
```
Should this become `torch::profiler` ?

aaronenyeshi reviewed (approved):
LGTM, thank you for this, I've glad we're moving the utils functions out of autograd into their own profiler folder.

Commit:
[Profiler] Pull helper methods into dedicated file. (And start `torch/csrc/profiler` folder. (#69255)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/69255

One thing that I've found as I optimize profier is that there's a lot of intermingled code, where the kineto profiler relies on the legacy (autograd) profiler for generic operations. This made optimization hard because I had to manage too many complex dependencies. (Exaserbated by the USE_KINETO #ifdef's sprinkled around.) This PR is the first of several to restructure the profiler(s) so the later optimizations go in easier.

Test Plan: Unit tests

Reviewed By: aaronenyeshi

Differential Revision: D32671972

fbshipit-source-id: 7c3b1b28f618f969ccd6318981f4089c98257904
